lib/ts/recipe/passwordless/utils.ts

```diff
@@ -41,19 +44,73 @@ export function validateAndNormaliseUserInput(
         ...config.override,
     };
 
+    let emailService =
+        config === undefined || config.emailDelivery === undefined
+            ? undefined
+            : undefined || config.emailDelivery.service;
+    /**
+     * following code is for backward compatibility.
+     * if user has not passed emailDelivery config, we
+     * use the createAndSendCustomEmail config. If the user
+     * has not passed even that config, we use the default
+     * createAndSendCustomEmail implementation
+     */
+    if (emailService === undefined) {
+        emailService = {
+            sendEmail: async (input: TypeEmailDeliveryTypeInput & { userContext: any }) => {
+                if (config.contactMethod === "EMAIL" || config.contactMethod === "EMAIL_OR_PHONE") {
+                    let createAndSendCustomEmail = config.createAndSendCustomEmail;
+                    if (createAndSendCustomEmail === undefined) {
+                        createAndSendCustomEmail = defaultCreateAndSendCustomEmail;
+                    }
+                    await createAndSendCustomEmail(
+                        {
+                            email: input.user.email,
+                            userInputCode: input.userInputCode,
+                            urlWithLinkCode: input.urlWithLinkCode,
+                            preAuthSessionId: input.preAuthSessionId,
+                            codeLifetime: input.codeLifetime,
+                        },
+                        input.userContext
+                    );
+                }
+            },
+        };
+    }
+
+    let emailDelivery = {
+        service: emailService,
+        override: (originalImplementation: EmailDeliveryIngredientInterface<TypeEmailDeliveryTypeInput>) =>
+            originalImplementation,
+        ...config.emailDelivery,
```

Comment:
You don't need to have this step here at all I think. And you can remove this from all recipes too?